### ISSUE TITLE
random_chain with same starting point (ex. tor)?

### ISSUE THREAD

**Issue by givemeurhats** [rofl0r/proxychains-ng#173]
Is it possible to do a random_chain but always have the same starting point (ex. tor)?

**Comment by tobwen** [comment]
Do it the easy way:
`proxychains4 -f tor.conf proxychains4 -f random.conf <command>`

**Comment by rofl0r** [comment]
i guess we could add a config item like RANDOM_CHAIN_FIXED_ENTRIES for that purpose. PR's welcome.

**Comment by tobwen** [comment]
Perhaps a more flexible approach would be useful?

```
[entry_chain]
socks4 127.0.0.1 9050

[random_chain]
socks4 192.0.1.1 9051
socks4 192.0.1.1 9052
socks4 192.0.1.1 9053

[static_chain]
socks5 x.x.x.x 9051
```
First go through _entry_chain_, than through one of _random_chain_ and through a _static_chain_ one again. You can do something like this in _3proxy_ already, but would be nice to have in _proxychains-ng_ :)

**Comment by rofl0r** [comment]
that would require incompatible config files.
the change i suggested can be implemented with minimal effort and changes, while this would require a lot of refactoring.

**Comment by tobwen** [comment]
Agreed. This just was a feature request to run through several chains ;)

**Comment by rofl0r** [comment]
hmm... since 3proxy can do this already, why not configure the chain sections via 3proxy, then run `proxychains4 -f connect_local_3proxy.conf myapp` ?

**Comment by tobwen** [comment]
That's what I am doing right now (the OT doesn't do perhaps, I've captured his thread).

But there's an annoying problem with 3proxy: It needs an even number of proxies to make its random chain working. Each chain needs a weight of 1000. I often have an odd number of proxies, which is a pitty ;) I've written a shell-script to calculate the weight for each proxy, but it's annoying...

Sure, this is a special case. For the requested TOR problem, it's a great solution!
